src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -267,8 +269,20 @@ protected internal int GetInvocationCounter()
         /// <returns>The next random number.</returns>
         protected internal BigInteger GetRandom()
         {
-            nonceData = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network);
-            return new BigInteger(nonceData, isUnsigned: true);
+            byte[] buffer;
+            long price;
+            if (IsHardforkEnabled(Hardfork.HF_Aspidochelone))
+            {
+                buffer = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network + random_times++);
```

Comment:
Now the first random number is the same as before. Is a hard fork still necessary?

Reply:
Transaction might request multiple random numbers, even if the first values are the same, the following random values are different. So, a `hardfork` is necessary.

Reply:
My question is, is there really a transaction that calls `GetRandom()` twice?

Reply:
> My question is, is there really a transaction that calls `GetRandom()` twice?

We can not know unless all contracts are open sourced. 

But from what i found on github, this method might call random twice: https://github.com/oespn/DAOCake-NEO/blob/a3047e12a8a27ad765d7a15b3cae38e35f241091/contracts/DAOCake/src/DAOCakeContract.cs#L118